### ISSUE TITLE
[redis 6] Authenticate with userpass from ACL

### ISSUE THREAD

**Issue by kodumbeats** [colinmollenhour/credis#158]
Redis 6 introduced userpass authentication with [ACLs](https://redis.io/topics/acl), which is used with managed Redis services offered by e.g. DigitalOcean. 

[phpredis](https://github.com/phpredis/phpredis#auth) supports this feature. What's the best way to support userpass auth in this library? 

I've got a crude [patch](https://github.com/colinmollenhour/credis/compare/master...kodumbeats:feat-auth-username) started but could benefit from some 👀 for setting up the test suite. :)

**Comment by Xon** [comment]
You should use `===` not `==` where possible. The strict checking means you don't need `!is_null($foo)`, it can just be `$foo !== null`

**Comment by Xon** [comment]
Type type hint should also be `string|null $password`.

Rather than hard-coding calling `call_user_func_array`, there is a switch statement for formatting arguments before being sent to phpredis.

Something like this should work; 
```php
                case 'zscan':
                    // allow phpredis to see the caller's reference
                    //$param_ref =& $args[0];
                    break;
                case 'auth':
                  $args = [$args];
                  break;
```

**Comment by Xon** [comment]
> Let's collaborate over a PR? I'm not sure how to set up the test suite to check ACL auth since that requires either a config or executing CLI commands: https://redis.io/topics/acl

I'ld recommend opening a PR, I think you've got it very close to done.

**Comment by colinmollenhour** [comment]
Thanks @kodumbeats and @Xon! Looking forward to PR.

**Comment by Xon** [comment]
@kodumbeats I've opened a PR with some fixes to the tests.

Mostly my changes are around the github tests and working on fixing the TLS support bug you discovered.

**Comment by Xon** [comment]
The TLS bug will need to be moved to another PR since getting the test working is proving to be trickier than expected.

**Comment by colinmollenhour** [comment]
Fixed in #161
